single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/svg/DefaultSVGWriter.java

```diff
@@ -1226,17 +1216,17 @@ private void drawNodesInfos(Element root, VoltageLevelGraph graph,
         double yPos = graph.getY() - layoutParameters.getVoltageLevelPadding().getTop() + graph.getHeight() + CIRCLE_RADIUS_NODE_INFOS_SIZE;
 
         double xShift = graph.getX() + xInitPos;
-        for (ElectricalNodeInfo node : initProvider.getElectricalNodesInfos(graph)) {
-            String idNode = metadata.getSvgParameters().getPrefixId() + "NODE_" + node.getBusId();
+        for (BusLegendInfo busLegendInfo : labelProvider.getNodesInfos(graph)) {
+            String idNode = metadata.getSvgParameters().getPrefixId() + "NODE_" + busLegendInfo.busId();
             Element gNode = nodesInfosNode.getOwnerDocument().createElement(GROUP);
             gNode.setAttribute("id", idNode);
 
-            List<String> styles = styleProvider.getBusStyles(node.getBusId(), graph);
-            drawNodeInfos(node, xShift, yPos, gNode, idNode, styles);
+            List<String> styles = styleProvider.getBusStyles(busLegendInfo.busId(), graph);
+            drawNodeInfos(busLegendInfo, xShift, yPos, gNode, idNode, styles);
 
             nodesInfosNode.appendChild(gNode);
 
-            metadata.addElectricalNodeInfoMetadata(new GraphMetadata.ElectricalNodeInfoMetadata(idNode, node.getUserDefinedId()));
+            metadata.addElectricalNodeInfoMetadata(new GraphMetadata.ElectricalNodeInfoMetadata(idNode));
```

Comment:
You forgot to replace it here also to be in line with your latest changes
```suggestion
            metadata.addBusLegendInfoMetadata(new GraphMetadata.BusLegendInfoMetadata(idNode));
```

Reply:
fixed